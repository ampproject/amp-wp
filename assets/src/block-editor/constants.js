export const TEXT_BLOCKS = [
	'core/paragraph',
	'core/heading',
	'core/code',
	'core/quote',
	'core/subhead',
];
