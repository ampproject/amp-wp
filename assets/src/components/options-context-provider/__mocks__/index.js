/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { createContext, useState } from '@wordpress/element';
/**
 * Internal dependencies
 */
import { READER } from '../../../common/constants';

export const Options = createContext();

/**
 * MOCK.
 *
 * @param {Object} props
 * @param {any}    props.children     Component children.
 * @param {string} props.themeSupport Default theme support mode in the mock provider.
 */
export function OptionsContextProvider({ children, themeSupport = READER }) {
	const [updates, updateOptions] = useState({});
	const [originalOptions, setOriginalOptions] = useState({
		mobile_redirect: true,
		theme_support: themeSupport,
	});

	return (
		<Options.Provider
			value={{
				editedOptions: { ...originalOptions, ...updates },
				originalOptions,
				setOriginalOptions,
				updates,
				updateOptions: (newOptions) => {
					updateOptions({ ...updates, newOptions });
				},
			}}
		>
			{children}
		</Options.Provider>
	);
}
OptionsContextProvider.propTypes = {
	children: PropTypes.any,
	themeSupport: PropTypes.string,
};
