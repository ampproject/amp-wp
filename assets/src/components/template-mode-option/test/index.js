/**
 * External dependencies
 */
import { act } from 'react-dom/test-utils';
import { create } from 'react-test-renderer';

/**
 * WordPress dependencies
 */
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { TemplateModeOption } from '..';
import { READER, STANDARD, TRANSITIONAL } from '../../../common/constants';
import { OptionsContextProvider } from '../../options-context-provider';

jest.mock('../../../components/options-context-provider');

let container;

describe('TemplateModeOption', () => {
	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

	it('matches snapshot', () => {
		let wrapper = create(
			<OptionsContextProvider>
				<TemplateModeOption
					mode={STANDARD}
					details="Standard info"
					detailsUrl="https://amp-wp.org/documentation/getting-started/standard/"
				>
					<div>
{'Component children'}
</div>
				</TemplateModeOption>
			</OptionsContextProvider>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();

		wrapper = create(
			<OptionsContextProvider>
				<TemplateModeOption
					mode={READER}
					previouslySelected={true}
					details="Reader info"
					detailsUrl="https://amp-wp.org/documentation/getting-started/reader/"
				/>
			</OptionsContextProvider>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();

		wrapper = create(
			<OptionsContextProvider>
				<TemplateModeOption
					details="Component details"
					detailsUrl="https://amp-wp.org/documentation/getting-started/reader/"
					mode={READER}
					previouslySelected={true}
					initialOpen={true}
					labelExtra={<div>
{'Extra label content'}
</div>}
				/>
			</OptionsContextProvider>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('is open by default if is current mode', () => {
		// Reader is the default in mock options context provider.
		act(() => {
			render(
				<OptionsContextProvider>
					<TemplateModeOption
						mode={READER}
						details="Reader info"
						detailsUrl="https://amp-wp.org/documentation/getting-started/reader/"
					>
						<div id="reader-mode-children">
{'children'}
</div>
					</TemplateModeOption>
					<TemplateModeOption
						mode={STANDARD}
						details="Standard info"
						detailsUrl="https://amp-wp.org/documentation/getting-started/standard/"
					>
						<div id="standard-mode-children">
{'children'}
</div>
					</TemplateModeOption>
				</OptionsContextProvider>,
				container
			);
		});

		expect(container.querySelector('#reader-mode-children')).not.toBeNull();
		expect(container.querySelector('#standard-mode-children')).toBeNull();
	});

	it('is open by default if initialOpen is true', () => {
		// Reader is the default in mock options context provider.
		act(() => {
			render(
				<OptionsContextProvider>
					<TemplateModeOption
						mode={TRANSITIONAL}
						details="Transitional info"
						detailsUrl="https://amp-wp.org/documentation/getting-started/transitional/"
					>
						<div id="reader-mode-children">
{'children'}
</div>
					</TemplateModeOption>
					<TemplateModeOption
						mode={STANDARD}
						initialOpen={true}
						details="Standard info"
						detailsUrl="https://amp-wp.org/documentation/getting-started/standard/"
					>
						<div id="standard-mode-children">
{'children'}
</div>
					</TemplateModeOption>
				</OptionsContextProvider>,
				container
			);
		});

		expect(
			container.querySelector('#transitional-mode-children')
		).toBeNull();
		expect(
			container.querySelector('#standard-mode-children')
		).not.toBeNull();
	});

	it('can be open', () => {
		act(() => {
			render(
				<OptionsContextProvider>
					<TemplateModeOption
						mode={TRANSITIONAL}
						initialOpen={true}
						details="Transitional info"
						detailsUrl="https://amp-wp.org/documentation/getting-started/transitional/"
					>
						<div id="transitional-mode-children">
{'children'}
</div>
					</TemplateModeOption>
				</OptionsContextProvider>,
				container
			);
		});

		expect(
			container.querySelector('#transitional-mode-children')
		).not.toBeNull();
	});

	it('can be closed', () => {
		act(() => {
			render(
				<OptionsContextProvider>
					<TemplateModeOption
						mode={TRANSITIONAL}
						initialOpen={false}
						details="Transitional info"
						detailsUrl="https://amp-wp.org/documentation/getting-started/transitional/"
					>
						<div id="transitional-mode-children">
{'children'}
</div>
					</TemplateModeOption>
				</OptionsContextProvider>,
				container
			);
		});

		expect(
			container.querySelector('#transitional-mode-children')
		).toBeNull();
	});
});
