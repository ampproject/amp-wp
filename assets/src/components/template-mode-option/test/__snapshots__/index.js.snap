// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateModeOption matches snapshot 1`] = `
<div
  className="amp-drawer amp-drawer--handle-type-right template-mode-option selectable selectable--left"
  id="template-mode-standard-container"
>
  <div
    className="amp-drawer__heading"
  >
    <label
      className="template-mode-option__label"
      htmlFor="template-mode-standard"
    >
      <div
        className="template-mode-selection__input-container"
      >
        <input
          checked={false}
          id="template-mode-standard"
          onChange={[Function]}
          type="radio"
        />
      </div>
      <div
        className="template-mode-selection__illustration"
      >
        <svg
          fill="none"
          height="58"
          viewBox="0 0 66 58"
          width="66"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.4008 9.85391C16.5008 9.85391 17.5008 8.95391 17.5008 7.75391C17.5008 6.55391 16.5008 5.75391 15.4008 5.75391C14.3008 5.75391 13.3008 6.65391 13.3008 7.85391C13.3008 9.05391 14.2008 9.85391 15.4008 9.85391Z"
            fill="#285BE7"
          />
          <path
            d="M22.1006 7.75391H25.1006"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M13.6006 37.3555H24.4006"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M29.8008 7.75391H32.8008"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M64.9004 1.75391H6.90039V45.9539H64.8004V1.75391H64.9004Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M60.4005 14.1523H13.0005V33.2523H60.4005V14.1523Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M19.8008 24.6523C21.2008 24.6523 22.3008 23.5523 22.3008 22.1523C22.3008 20.7523 21.2008 19.6523 19.8008 19.6523C18.4008 19.6523 17.3008 20.7523 17.3008 22.1523C17.3008 23.5523 18.4008 24.6523 19.8008 24.6523Z"
            fill="#285BE7"
          />
          <path
            d="M23.6006 28.0523L32.0006 22.0523L38.0006 25.4523L44.9006 20.1523V28.1523C44.9006 28.0523 25.1006 28.0523 23.6006 28.0523Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M8.3542 57.5023C12.6542 57.5023 16.1542 54.0023 16.1542 49.7023C16.1542 45.4023 12.6542 41.9023 8.3542 41.9023C4.0542 41.9023 0.554199 45.4023 0.554199 49.7023C0.554199 54.0023 4.0542 57.5023 8.3542 57.5023Z"
            fill="#3363E8"
          />
          <path
            d="M9.35446 45.2992L8.95446 48.6992H10.4545C10.4545 48.6992 11.0545 48.5992 10.4545 49.5992C7.75446 54.0992 7.75446 54.0992 7.75446 54.0992H7.25446L7.75446 50.6992H6.45446C6.45446 50.6992 5.45446 50.9992 6.15446 49.8992C8.95446 45.4992 9.05446 45.1992 9.05446 45.1992L9.35446 45.2992Z"
            fill="white"
          />
        </svg>
      </div>
      <div
        className="template-mode-selection__description"
      >
        <h3>
          Standard
        </h3>
      </div>
    </label>
  </div>
  <div
    className="components-panel__body amp-drawer__panel-body"
  >
    <h2
      className="components-panel__body-title"
    >
      <button
        aria-describedby={null}
        aria-expanded={false}
        className="components-button components-panel__body-toggle"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          <svg
            aria-hidden={true}
            className="components-panel__arrow"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </span>
        <span
          className="components-visually-hidden"
        >
          Standard
        </span>
      </button>
    </h2>
  </div>
</div>
`;

exports[`TemplateModeOption matches snapshot 2`] = `
<div
  className="amp-drawer amp-drawer--handle-type-right template-mode-option amp-drawer--opened selectable selectable--selected selectable--left"
  id="template-mode-reader-container"
>
  <div
    className="amp-drawer__heading"
  >
    <label
      className="template-mode-option__label"
      htmlFor="template-mode-reader"
    >
      <div
        className="template-mode-selection__input-container"
      >
        <input
          checked={true}
          id="template-mode-reader"
          onChange={[Function]}
          type="radio"
        />
      </div>
      <div
        className="template-mode-selection__illustration"
      >
        <svg
          fill="none"
          height="63"
          viewBox="0 0 78 63"
          width="78"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.2068 37.5117H24.0068"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M14.9067 10.0102C16.0067 10.0102 17.0067 9.11016 17.0067 7.91016C17.0067 6.71016 16.1067 5.91016 14.9067 5.91016C13.8067 5.91016 12.9067 6.81016 12.9067 7.91016C12.9067 9.01016 13.8067 10.0102 14.9067 10.0102Z"
            fill="#285BE7"
          />
          <path
            d="M21.6069 7.91016H24.6069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M29.4067 7.91016H32.4067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M64.4068 1.91016H6.50684V46.1102H64.4068V1.91016Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M60.0069 14.3086H12.6069V33.4086H60.0069V14.3086Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M19.3069 24.8086C20.7069 24.8086 21.8069 23.7086 21.8069 22.3086C21.8069 20.9086 20.7069 19.8086 19.3069 19.8086C17.9069 19.8086 16.8069 20.9086 16.8069 22.3086C16.8069 23.7086 17.9069 24.8086 19.3069 24.8086Z"
            fill="#285BE7"
          />
          <path
            d="M23.2068 28.2086L31.6068 22.2086L37.6068 25.6086L44.5068 20.3086V28.3086C44.5068 28.2086 24.7068 28.2086 23.2068 28.2086Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M69.6069 17.6094H48.3069V54.3094H69.5069V17.6094H69.6069Z"
            fill="white"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M69.7068 17.6094H48.5068V28.4094H69.7068V17.6094Z"
            fill="#285BE7"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M59.6069 23.3086H62.6069"
            stroke="white"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.8068 25.3094C54.9068 25.3094 55.9068 24.4094 55.9068 23.2094C55.9068 22.1094 55.0068 21.1094 53.8068 21.1094C52.7068 21.1094 51.7068 22.0094 51.7068 23.2094C51.7068 24.4094 52.7068 25.3094 53.8068 25.3094Z"
            fill="white"
          />
          <path
            d="M56.9067 33.9102H65.1067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.1069 33.9102H53.4069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M56.9067 37.3086H65.1067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.1069 37.3086H53.4069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M56.9067 41.0117H65.1067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.1069 41.0117H53.4069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M69.2192 62.1117C73.5192 62.1117 77.0192 58.6117 77.0192 54.3117C77.0192 50.0117 73.5192 46.5117 69.2192 46.5117C64.9192 46.5117 61.4192 50.0117 61.4192 54.3117C61.4192 58.6117 64.9192 62.1117 69.2192 62.1117Z"
            fill="#3363E8"
          />
          <path
            d="M70.2194 49.9086L69.8194 53.3086H71.3194C71.3194 53.3086 71.9194 53.2086 71.3194 54.2086C68.6194 58.7086 68.6194 58.7086 68.6194 58.7086H68.1194L68.6194 55.3086H67.3194C67.3194 55.3086 66.3194 55.6086 67.0194 54.5086C69.8194 50.1086 69.9194 49.8086 69.9194 49.8086L70.2194 49.9086Z"
            fill="white"
          />
          <circle
            cx="8.16113"
            cy="49.8125"
            fill="white"
            r="8"
          />
          <circle
            cx="8.16113"
            cy="49.8125"
            fill="white"
            r="8"
          />
          <path
            d="M8.16089 41.8125C3.76089 41.8125 0.160889 45.4125 0.160889 49.8125C0.160889 54.2125 3.76089 57.8125 8.16089 57.8125C12.5609 57.8125 16.1609 54.2125 16.1609 49.8125C16.1609 45.4125 12.5609 41.8125 8.16089 41.8125ZM13.6809 46.6125H11.3609C11.1209 45.6525 10.7209 44.6925 10.2409 43.7325C11.6809 44.2925 12.9609 45.2525 13.6809 46.6125ZM8.16089 43.4125C8.80089 44.3725 9.36089 45.4125 9.68089 46.6125H6.64089C6.96089 45.4925 7.52089 44.3725 8.16089 43.4125ZM2.00089 51.4125C1.84089 50.9325 1.76089 50.3725 1.76089 49.8125C1.76089 49.2525 1.84089 48.6925 2.00089 48.2125H4.72089C4.64089 48.7725 4.64089 49.2525 4.64089 49.8125C4.64089 50.3725 4.72089 50.8525 4.72089 51.4125H2.00089ZM2.64089 53.0125H5.04089C5.28089 53.9725 5.68089 55.0125 6.16089 55.8925C4.64089 55.3325 3.36089 54.3725 2.64089 53.0125ZM4.96089 46.6125H2.56089C3.36089 45.2525 4.56089 44.2925 6.00089 43.7325C5.60089 44.6925 5.28089 45.6525 4.96089 46.6125ZM8.16089 56.2125C7.52089 55.2525 6.96089 54.2125 6.64089 53.0125H9.68089C9.36089 54.1325 8.80089 55.2525 8.16089 56.2125ZM10.0009 51.4125H6.32089C6.24089 50.8525 6.16089 50.3725 6.16089 49.8125C6.16089 49.2525 6.24089 48.6925 6.32089 48.2125H10.0809C10.1609 48.6925 10.2409 49.2525 10.2409 49.8125C10.2409 50.3725 10.0809 50.8525 10.0009 51.4125ZM10.2409 55.8925C10.7209 55.0125 11.1209 54.0525 11.3609 53.0125H13.6809C12.9609 54.2925 11.6809 55.3325 10.2409 55.8925ZM11.6809 51.4125C11.7609 50.8525 11.7609 50.3725 11.7609 49.8125C11.7609 49.2525 11.6809 48.7725 11.6809 48.2125H14.4009C14.5609 48.6925 14.6409 49.2525 14.6409 49.8125C14.6409 50.3725 14.5609 50.9325 14.4009 51.4125H11.6809Z"
            fill="#285BE7"
          />
        </svg>
      </div>
      <div
        className="template-mode-selection__description"
      >
        <h3>
          Reader
        </h3>
        <div
          className="amp-info"
        >
          Previously selected
        </div>
      </div>
    </label>
  </div>
  <div
    className="components-panel__body amp-drawer__panel-body is-opened"
  >
    <h2
      className="components-panel__body-title"
    >
      <button
        aria-describedby={null}
        aria-expanded={true}
        className="components-button components-panel__body-toggle"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          <svg
            aria-hidden={true}
            className="components-panel__arrow"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.5 12.4L12 8l5.5 4.4-.9 1.2L12 10l-4.5 3.6-1-1.2z"
            />
          </svg>
        </span>
        <span
          className="components-visually-hidden"
        >
          Reader
        </span>
      </button>
    </h2>
    <div
      className="amp-drawer__panel-body-inner"
    >
      <div
        className="template-mode-selection__details"
      >
        <p>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Reader info",
              }
            }
          />
           
          <a
            href="https://amp-wp.org/documentation/getting-started/reader/"
            rel="noreferrer noopener"
            target="_blank"
          >
            Learn more.
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`TemplateModeOption matches snapshot 3`] = `
<div
  className="amp-drawer amp-drawer--handle-type-right template-mode-option amp-drawer--opened selectable selectable--selected selectable--left"
  id="template-mode-reader-container"
>
  <div
    className="amp-drawer__heading"
  >
    <label
      className="template-mode-option__label"
      htmlFor="template-mode-reader"
    >
      <div
        className="template-mode-selection__input-container"
      >
        <input
          checked={true}
          id="template-mode-reader"
          onChange={[Function]}
          type="radio"
        />
      </div>
      <div
        className="template-mode-selection__illustration"
      >
        <svg
          fill="none"
          height="63"
          viewBox="0 0 78 63"
          width="78"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.2068 37.5117H24.0068"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M14.9067 10.0102C16.0067 10.0102 17.0067 9.11016 17.0067 7.91016C17.0067 6.71016 16.1067 5.91016 14.9067 5.91016C13.8067 5.91016 12.9067 6.81016 12.9067 7.91016C12.9067 9.01016 13.8067 10.0102 14.9067 10.0102Z"
            fill="#285BE7"
          />
          <path
            d="M21.6069 7.91016H24.6069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M29.4067 7.91016H32.4067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M64.4068 1.91016H6.50684V46.1102H64.4068V1.91016Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M60.0069 14.3086H12.6069V33.4086H60.0069V14.3086Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M19.3069 24.8086C20.7069 24.8086 21.8069 23.7086 21.8069 22.3086C21.8069 20.9086 20.7069 19.8086 19.3069 19.8086C17.9069 19.8086 16.8069 20.9086 16.8069 22.3086C16.8069 23.7086 17.9069 24.8086 19.3069 24.8086Z"
            fill="#285BE7"
          />
          <path
            d="M23.2068 28.2086L31.6068 22.2086L37.6068 25.6086L44.5068 20.3086V28.3086C44.5068 28.2086 24.7068 28.2086 23.2068 28.2086Z"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M69.6069 17.6094H48.3069V54.3094H69.5069V17.6094H69.6069Z"
            fill="white"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M69.7068 17.6094H48.5068V28.4094H69.7068V17.6094Z"
            fill="#285BE7"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M59.6069 23.3086H62.6069"
            stroke="white"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.8068 25.3094C54.9068 25.3094 55.9068 24.4094 55.9068 23.2094C55.9068 22.1094 55.0068 21.1094 53.8068 21.1094C52.7068 21.1094 51.7068 22.0094 51.7068 23.2094C51.7068 24.4094 52.7068 25.3094 53.8068 25.3094Z"
            fill="white"
          />
          <path
            d="M56.9067 33.9102H65.1067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.1069 33.9102H53.4069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M56.9067 37.3086H65.1067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.1069 37.3086H53.4069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M56.9067 41.0117H65.1067"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M53.1069 41.0117H53.4069"
            stroke="#285BE7"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
          <path
            d="M69.2192 62.1117C73.5192 62.1117 77.0192 58.6117 77.0192 54.3117C77.0192 50.0117 73.5192 46.5117 69.2192 46.5117C64.9192 46.5117 61.4192 50.0117 61.4192 54.3117C61.4192 58.6117 64.9192 62.1117 69.2192 62.1117Z"
            fill="#3363E8"
          />
          <path
            d="M70.2194 49.9086L69.8194 53.3086H71.3194C71.3194 53.3086 71.9194 53.2086 71.3194 54.2086C68.6194 58.7086 68.6194 58.7086 68.6194 58.7086H68.1194L68.6194 55.3086H67.3194C67.3194 55.3086 66.3194 55.6086 67.0194 54.5086C69.8194 50.1086 69.9194 49.8086 69.9194 49.8086L70.2194 49.9086Z"
            fill="white"
          />
          <circle
            cx="8.16113"
            cy="49.8125"
            fill="white"
            r="8"
          />
          <circle
            cx="8.16113"
            cy="49.8125"
            fill="white"
            r="8"
          />
          <path
            d="M8.16089 41.8125C3.76089 41.8125 0.160889 45.4125 0.160889 49.8125C0.160889 54.2125 3.76089 57.8125 8.16089 57.8125C12.5609 57.8125 16.1609 54.2125 16.1609 49.8125C16.1609 45.4125 12.5609 41.8125 8.16089 41.8125ZM13.6809 46.6125H11.3609C11.1209 45.6525 10.7209 44.6925 10.2409 43.7325C11.6809 44.2925 12.9609 45.2525 13.6809 46.6125ZM8.16089 43.4125C8.80089 44.3725 9.36089 45.4125 9.68089 46.6125H6.64089C6.96089 45.4925 7.52089 44.3725 8.16089 43.4125ZM2.00089 51.4125C1.84089 50.9325 1.76089 50.3725 1.76089 49.8125C1.76089 49.2525 1.84089 48.6925 2.00089 48.2125H4.72089C4.64089 48.7725 4.64089 49.2525 4.64089 49.8125C4.64089 50.3725 4.72089 50.8525 4.72089 51.4125H2.00089ZM2.64089 53.0125H5.04089C5.28089 53.9725 5.68089 55.0125 6.16089 55.8925C4.64089 55.3325 3.36089 54.3725 2.64089 53.0125ZM4.96089 46.6125H2.56089C3.36089 45.2525 4.56089 44.2925 6.00089 43.7325C5.60089 44.6925 5.28089 45.6525 4.96089 46.6125ZM8.16089 56.2125C7.52089 55.2525 6.96089 54.2125 6.64089 53.0125H9.68089C9.36089 54.1325 8.80089 55.2525 8.16089 56.2125ZM10.0009 51.4125H6.32089C6.24089 50.8525 6.16089 50.3725 6.16089 49.8125C6.16089 49.2525 6.24089 48.6925 6.32089 48.2125H10.0809C10.1609 48.6925 10.2409 49.2525 10.2409 49.8125C10.2409 50.3725 10.0809 50.8525 10.0009 51.4125ZM10.2409 55.8925C10.7209 55.0125 11.1209 54.0525 11.3609 53.0125H13.6809C12.9609 54.2925 11.6809 55.3325 10.2409 55.8925ZM11.6809 51.4125C11.7609 50.8525 11.7609 50.3725 11.7609 49.8125C11.7609 49.2525 11.6809 48.7725 11.6809 48.2125H14.4009C14.5609 48.6925 14.6409 49.2525 14.6409 49.8125C14.6409 50.3725 14.5609 50.9325 14.4009 51.4125H11.6809Z"
            fill="#285BE7"
          />
        </svg>
      </div>
      <div
        className="template-mode-selection__description"
      >
        <h3>
          Reader
        </h3>
        <div
          className="amp-info"
        >
          Previously selected
        </div>
        <div
          className="template-mode-selection__label-extra"
        >
          <div>
            Extra label content
          </div>
        </div>
      </div>
    </label>
  </div>
  <div
    className="components-panel__body amp-drawer__panel-body is-opened"
  >
    <h2
      className="components-panel__body-title"
    >
      <button
        aria-describedby={null}
        aria-expanded={true}
        className="components-button components-panel__body-toggle"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          <svg
            aria-hidden={true}
            className="components-panel__arrow"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.5 12.4L12 8l5.5 4.4-.9 1.2L12 10l-4.5 3.6-1-1.2z"
            />
          </svg>
        </span>
        <span
          className="components-visually-hidden"
        >
          Reader
        </span>
      </button>
    </h2>
    <div
      className="amp-drawer__panel-body-inner"
    >
      <div
        className="template-mode-selection__details"
      >
        <p>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Component details",
              }
            }
          />
           
          <a
            href="https://amp-wp.org/documentation/getting-started/reader/"
            rel="noreferrer noopener"
            target="_blank"
          >
            Learn more.
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
